public/app/features/live/pages/SettingsEditor.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from 'react';
+import { CodeEditor, Select } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { Rule, Setting, PipelineListOption } from './types';
+
+interface Props {
+  rule: Rule;
+  types: Array<SelectableValue<string>>;
+  onBlur: (text: string, setting: Setting, settingType: string | undefined) => void;
+  setting: Setting;
+  pipelineEntitiesList: [];
+}
+
+const SettingsEditor: React.FC<Props> = ({ rule, onBlur, types, setting, pipelineEntitiesList }) => {
+  const settingBody = rule.settings[setting];
+  let type = settingBody?.type;
+  let settingValue = {};
+  const [settingType, setSettingType] = useState<string | undefined>(type);
+
+  useEffect(() => {
+    if (type) {
+      setSettingType(type);
+    }
+  }, [type]);
+  // get the example body of the type out
+  const example = pipelineEntitiesList?.filter((option: PipelineListOption) => option.type === settingType)?.[0]?.[
```

Comment:
I think trying to set this as part of render is confusing things with change events...